dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/programrule/implementers/enrollment/ShowErrorWarningExecutorTest.java

```diff
@@ -0,0 +1,173 @@
+        assertTrue( error.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenAShowErrorActionIsTriggeredForCompletedEnrollment()
+    {
+        Optional<ProgramRuleIssue> error = showErrorExecutor.executeEnrollmentRuleAction( bundle,
+            completedEnrollment() );
+        assertTrue( error.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAWarningWhenAShowErrorActionIsTriggeredForActiveEnrollment()
+    {
+        Optional<ProgramRuleIssue> warning = showWarningExecutor.executeEnrollmentRuleAction( bundle,
+            activeEnrollment() );
+        assertTrue( warning.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAWarningWhenAShowErrorActionIsTriggeredForCompletedEnrollment()
+    {
+        Optional<ProgramRuleIssue> warning = showWarningExecutor.executeEnrollmentRuleAction( bundle,
+            completedEnrollment() );
+        assertTrue( warning.isPresent() );
+    }
+
+    @Test
+    void shouldNotReturnAnErrorWhenAShowErrorOnCompleteActionIsTriggeredForActiveEnrollment()
+    {
+        Optional<ProgramRuleIssue> error = errorOnCompleteExecutor.executeEnrollmentRuleAction( bundle,
+            activeEnrollment() );
+        assertFalse( error.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAnErrorWhenAShowErrorOnCompleteActionIsTriggeredForCompletedEnrollment()
+    {
+        Optional<ProgramRuleIssue> error = errorOnCompleteExecutor.executeEnrollmentRuleAction( bundle,
+            completedEnrollment() );
+        assertTrue( error.isPresent() );
+    }
+
+    @Test
+    void shouldNotReturnAWarningWhenAShowErrorOnCompleteActionIsTriggeredForActiveEnrollment()
+    {
+        Optional<ProgramRuleIssue> warning = warningOnCompleteExecutor.executeEnrollmentRuleAction( bundle,
+            activeEnrollment() );
+        assertFalse( warning.isPresent() );
+    }
+
+    @Test
+    void shouldReturnAWarningWhenAShowErrorOnCompleteActionIsTriggeredForCompletedEnrollment()
+    {
+        Optional<ProgramRuleIssue> warning = warningOnCompleteExecutor.executeEnrollmentRuleAction( bundle,
+            completedEnrollment() );
+        assertTrue( warning.isPresent() );
+    }
+
+    private List<Enrollment> getEnrollments()
+    {
+        return Lists.newArrayList( activeEnrollment(), completedEnrollment() );
+    }
+
+    private Enrollment activeEnrollment()
+    {
+        Enrollment activeEnrollment = new Enrollment();
+        activeEnrollment.setEnrollment( ACTIVE_ENROLLMENT_ID );
+        activeEnrollment.setStatus( EnrollmentStatus.ACTIVE );
+        return activeEnrollment;
+    }
+
+    private Enrollment completedEnrollment()
+    {
+        Enrollment completedEnrollment = new Enrollment();
+        completedEnrollment.setEnrollment( COMPLETED_ENROLLMENT_ID );
+        completedEnrollment.setStatus( EnrollmentStatus.COMPLETED );
+        return completedEnrollment;
+    }
+
+    private ErrorWarningRuleAction getErrorActionRule()
```

Comment:
This is a bit confusing to me 🤔 

Return type: ErrorWarningRuleAction
method name: getErrorActionRule
usages for example:
  * `new ShowErrorExecutor( getErrorActionRule() );`
  * `new ShowWarningExecutor( getErrorActionRule() );`
 
I see a mix of error and warning in names and types. So I assume `ErrorWarningRuleAction` is like a sum type containing either an error or a warning rule.

Does it matter for an `Executor` whether its an error or warning action rule?